### ISSUE TITLE
shared_preferences do not support armv7

### ISSUE THREAD

**Issue by bernaferrari** [flutter/flutter#76015]
I made an app for iOS and it crashes in arm7 devices. Upon further inspection, it seems the cause is shared_preferences: `.../build/ios/Debug-iphoneos/Runner.app/Frameworks/shared_preferences.framewo
    rk/shared_preferences is not armv7.`

Therefore, it seems I can't support old iOS devices if I use shared_preferences.

```
   sent 50854373 bytes  received 198 bytes  33903047.33 bytes/sec
    total size is 51848077  speedup is 1.02
    building file list ... done
    Flutter.framework/
    Flutter.framework/Flutter
    Flutter.framework/Info.plist
    Flutter.framework/icudtl.dat
    Flutter.framework/_CodeSignature/
    Flutter.framework/_CodeSignature/CodeResources

    sent 66243816 bytes  received 120 bytes  132487872.00 bytes/sec
    total size is 66235305  speedup is 1.00
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Frameworks/App.framework/App: signed
    bundle with Mach-O thin (armv7) [io.flutter.flutter.app]
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Frameworks/Flutter.framework/Flutter:
    signed bundle with Mach-O universal (armv7 arm64) [io.flutter.flutter]
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Info.plist: Could not extract value,
    error: No value at that key path or invalid key path: NSBonjourServices
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Info.plist: Could not extract value,
    error: No value at that key path or invalid key path: NSLocalNetworkUsageDescription
    Non-fat binary
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Frameworks/shared_preferences.framewo
    rk/shared_preferences is not armv7. Running lipo -info:
    Non-fat file:
    /Users/bernardoferrari/Documents/colorstudio/build/ios/Debug-iphoneos/Runner.app/Frameworks/shared_preferences.framewo
    rk/shared_preferences is architecture: arm64
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the precompiled application for the device.
```

**Comment by darshankawar** [comment]
@bernaferrari 
Can you provide `flutter doctor -v` to know iOS version of your device ?
Also, can you try to change workspace settings to Legacy and see if it helps ?

**Comment by bernaferrari** [comment]
I had an iPad with iOS 10 and a friend had another 32bit iPad, which was crashing when my app was released to the App Store. Upon getting the logs, it seems shared_preferences is arm64 only. I couldn't check Legacy yet.

**Comment by darshankawar** [comment]
Thanks @bernaferrari for the details. Does it crash locally in non-prod environment too ? Is it possible to get a reproducible use case ?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
